File: mobile_app/app/src/main/java/com/example/licenta_mobile/rest/RegistrationService.java
package com.example.licenta_mobile.rest;

import com.example.licenta_mobile.dto.MessageDto;
import com.example.licenta_mobile.dto.ParkingPlaceDto;
import com.example.licenta_mobile.dto.RegistrationDto;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;

public interface RegistrationService {
    @POST( "/register" )
    @Headers( {
            "Content-Type: application/json"
    } )
    Call<MessageDto> register (@Body RegistrationDto registrationDto);
}

File: mobile_app/app/src/main/java/com/example/licenta_mobile/rest/RegistrationService.kt
package com.example.licenta_mobile.rest

import com.example.licenta_mobile.dto.RegistrationDto
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

interface RegistrationService {

    @POST("/register")
    fun register(@Body registrationDto: RegistrationDto): Call<Int>
}